modin/core/execution/dispatching/factories/test/test_dispatcher.py

```diff
@@ -114,8 +114,9 @@ def test_factory_switch():
 
 def test_engine_wrong_factory():
     with pytest.raises(FactoryNotFoundError):
-        with _switch_value(Engine, "BadEngine"):
```

Comment:
Now, because we added `Engine.subscribe(_update_engine)` into `get_factory`, there will be another error here (about the non-existence of the engine). To reproduce `FactoryNotFoundError` problem, we can make a combination of engine and storage format, which we do not have.